### PR TITLE
Elastichsearch stateful sets, Readme related changes, Intermediate Bazel upgrade 

### PR DESCRIPTION
**Pull request by mencywoo**
## Highlight of changes
- Remove es-master PV during teardown to avoid orphaned disks left behind 
- Provision more CPU's for ES cluster. The original n1-standard-2 leads to failure to provision the 3rd data pod due to resource constraints and anti-affinity.
- README.md, where software requirements have been expanded and deprecated instructions are removed
- Rename some scripts to align with the README.md instructions.
- Minimal Bazel upgrade from 0.20 to 0.21 such that CI can run 

## Note
- I have moved the regions to us-central1 to work around issues in us-west1. Happy to revert if us-west1 is considered optimal for demo
- Bazel 0.21.0 is the last compatible version to existing code.  Further upgrade will require more indepth analysis,  thus not included in this PR.

### CODE REVIEW

File: scripts/on-prem-destroy.sh [hunk 1]
```diff
@@ -52,8 +52,18 @@ sleep "$(( GRACE + PADDING ))"
...
 kubectl --namespace default delete pvc -l component=elasticsearch,role=data || true
+kubectl --namespace default delete pvc -l component=elasticsearch,role=master || true
 
-echo "kubectl get pvc --all-namespaces"
-kubectl get pvc --all-namespaces
-echo "kubectl get pv --all-namespaces"
-kubectl get pv --all-namespaces
+timeout=100
+
+# Attempt to wait for the PV to be destroyed before proceeding. Most of the time the PV's are deleted immediately,
```
Comment (chrislovecnm) [inline scripts/on-prem-destroy.sh]: PV's are not deleted immediately with stateful sets.  That is why we have to clean them up.  Please update the comments.

File: scripts/teardown.sh [hunk 2]
```diff
@@ -25,6 +25,7 @@
 set -o errexit
 # set -o nounset
 set -o pipefail
+set -x
```
Comment (chrislovecnm) [inline scripts/teardown.sh]: why?

File: scripts/on-prem-destroy.sh (hunk 1 above)
Comment (mencywoo) [reply scripts/on-prem-destroy.sh]: comment updated

File: scripts/teardown.sh (hunk 2 above)
Comment (mencywoo) [reply scripts/teardown.sh]: removed the debugging flag

**Comment by cxhercules** [comment]
Please test this

**Comment by mencywoo** [comment]
Need to make a number of changes to satisfy requirement of CI pipeline, resulting in Bazel partial upgrade

File: scripts/cloud-destroy.sh [hunk 3]
```diff
@@ -37,12 +37,16 @@ if [[ ! $(kubectl config use-context "$CONTEXT") ]]; then
 fi
 
 # delete k8s resources
+# Note:  the Bazel flag incompatible_package_name_is_a_function, while necessary for a successful build, is only valid in Bazel v0.21.
+# This flag is deprecated in Bazel v0.22+
 bazel run \
+  --incompatible_package_name_is_a_function=false \
```
Comment (chrislovecnm) [inline scripts/cloud-destroy.sh]: backlog removing this please

**Pull request merged**